provider/resource_service.go

```diff
@@ -402,17 +394,17 @@ func (r *resourceService) ModifyPlan(ctx context.Context, req resource.ModifyPla
 		if existingKpi, ok := kpiOldKeys[internalIdentifier]; ok {
 			kpi.ID = existingKpi.ID
 			kpi.Urgency = existingKpi.Urgency
-		}
-
-		properties := []*types.String{
-			&kpi.Description, &kpi.ThresholdTemplateID,
-		}
-
-		for _, p := range properties {
-			if p.IsUnknown() {
-				*p = types.StringNull()
+			if kpi.Urgency.IsUnknown() {
+				kpi.Urgency = existingKpi.Urgency
```

Comment:
this `if` seems unnecessary because you are setting Urgency the same way one the line above(396)

Reply:
oh, thank you, that should be  `=default_urgency`

Reply:
its above, yes, just unnecessary